keyfreq.sh

```diff
@@ -1,32 +1,39 @@
 #!/bin/bash
 
 
-# logs the key press frequency over 9 second window. Logs are written 
+# logs the key press frequency over 9 second window. Logs are written
 # in logs/keyfreqX.txt every 9 seconds, where X is unix timestamp of 7am of the
 # recording day.
 
 LANG=en_US.utf8
 
-helperfile="logs/keyfreqraw.txt" # temporary helper file
+helperfile='/dev/shm/keyfreqraw.txt'
 
 mkdir -p logs
 
+trap 'kill $(jobs -p)' EXIT
+
 while true
 do
-  showkey > $helperfile &
-  PID=$!
-  
-  # work in windows of 9 seconds 
+  xinput test 11 | tr -d '0-9' > $helperfile &
```

Comment:
Makes sense, done. There might be other ways to configure this for example by including a config file. I just went with environment variables for now.